toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/StartingPointDetailsScreen.kt

```diff
@@ -17,72 +17,288 @@
-                modifier = Modifier.clickable { onBackPressed() },
-                horizontalArrangement = Arrangement.SpaceBetween,
-                verticalAlignment = Alignment.CenterVertically
-            ) {
-            Image(
-                imageVector = Icons.AutoMirrored.Default.ArrowBack,
-                contentDescription = "back"
+
+            BackButton(onBackPressed)
+
+            Spacer(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .height(10.dp)
             )
-            ReadOnlyTextField(
-                text = stringResource(id = R.string.starting_points))
-            }
-        }
 
-        startingPoint?.let { point ->
-            val geometry = point.feature.geometry
+            Title(startingPoint.name, onZoomTo, onDelete)
 
-            if (geometry != null && geometry is Polyline ) {
-                FractionAlongEdgeSlider(point, onFractionChanged)
+            Spacer(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .height(10.dp)
+            )
+
+            LazyColumn(modifier = Modifier
+                .padding(vertical = 3.dp),
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                if (startingPoint.utilityElement.networkSource.sourceType == UtilityNetworkSourceType.Edge) {
+                    item {
+                        FractionAlongEdgeSlider(startingPoint, onFractionChanged)
+                    }
+                } else if (startingPoint.utilityElement.networkSource.sourceType == UtilityNetworkSourceType.Junction
+                    && startingPoint.utilityElement.terminal != null
+                    && !startingPoint.utilityElement.assetType.terminalConfiguration?.terminals.isNullOrEmpty()
+                ) {
+                    item {
+                        TerminalConfiguration(startingPoint, onTerminalSelected)
+                    }
+                }
+                item {
+                    Attributes(startingPoint)
+                }
             }
         }
+    }
+}
 
+@Composable
+private fun BackButton(
+    onBackPressed: () -> Unit
+) {
+    Row(
+        modifier = Modifier.clickable { onBackPressed() },
+        horizontalArrangement = Arrangement.SpaceBetween,
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Image(
+            modifier = Modifier.padding(start = 12.dp),
+            imageVector = Icons.AutoMirrored.Sharp.ArrowBack,
+            contentDescription = "back"
+        )
+        Text(
+            modifier = Modifier.padding(start = 8.dp),
+            text = stringResource(id = R.string.starting_points),
+            style = MaterialTheme.typography.titleMedium,
+            maxLines = 1,
+            overflow = TextOverflow.Ellipsis
+        )
     }
 }
 
 @Composable
 private fun FractionAlongEdgeSlider(startingPoint: StartingPoint, onFractionChanged: (StartingPoint, Float) -> Unit) {
```

Comment:
Nitpick, you don't need to pass `startingPoint` here if you're just going to pass it back up again in your `onValueChanged`. You can hoist it up. 
```suggestion
private fun FractionAlongEdgeSlider(fractionAlongEdge: Float, onFractionChanged: (Float) -> Unit)
```

Reply:
I don't see the advantage of doing this, I like the function call being simple and the logic needed is contained in the fun displaying the data.